webapp/components/form/Dropdown/Dropdown.js

```diff
@@ -1,143 +1,96 @@
     selection,
-    sourceElement,
+    testId,
     title,
     validation,
   } = props
 
-  const dropdownRef = useRef(null)
-  const inputRef = useRef(null)
-
-  const itemLabelFunction = useCallback(
-    (item) => (itemLabel.constructor === String ? A.prop(itemLabel, item) : itemLabel(item)),
-    [itemLabel]
-  )
-
-  const { state, Actions } = useLocalState({
-    autocompleteMinChars,
-    disabled,
-    itemKey,
-    itemLabelFunction,
-    items,
-    onBeforeChange,
-    onChange,
-    readOnly,
-    selection,
-    title,
-  })
-
-  // update itemLabelFunction and input value on itemLabelFunction change
-  useEffect(() => {
-    Actions.updateItemLabelFunction({ itemLabelFunction, selection })
-  }, [itemLabelFunction])
-
-  const showDialog = State.getShowDialog(state)
-  const itemsDialog = State.getItemsDialog(state)
-  const inputValue = State.getInputValue(state) || ''
-
-  const searchMinCharsReached = autocompleteMinChars <= 0 || inputValue.trim().length >= autocompleteMinChars
+  const { inputId, loading, menuIsOpen, onChange, onInputChange, openMenuOnClick, options, selectRef, value } =
+    useDropdown({
+      autocompleteMinChars,
+      idInputProp,
+      itemKey,
+      itemLabel,
+      itemsProp,
+      onBeforeChange,
+      onChangeProp,
+      selection,
+      title,
+    })
 
   return (
-    <div
-      ref={dropdownRef}
-      className={classNames('dropdown', className)}
-      onBlur={(event) => {
-        const { className: classNameTarget = '' } = event.relatedTarget || {}
-        if (classNameTarget !== ItemDialog.className) {
-          Actions.closeDialog({ selection })
-        }
-      }}
+    <ValidationTooltip
+      className="dropdown-wrapper dropdown-validation-tooltip"
+      id={id}
+      testId={testId}
+      validation={validation}
     >
-      <Input
-        id={idInput}
-        ref={inputRef}
+      <ReactSelect
+        className={classNames('dropdown', className)}
+        classNamePrefix="dropdown"
+        components={{ Option: OptionComponent, IndicatorsContainer: IndicatorsContainerComponent }}
+        inputId={inputId}
+        isClearable={clearable && !readOnly}
+        isDisabled={disabled}
+        isLoading={loading}
+        isSearchable={!readOnlyInput && !readOnly}
+        onChange={onChange}
+        openMenuOnClick={openMenuOnClick}
+        menuIsOpen={menuIsOpen}
+        onInputChange={onInputChange}
+        options={options}
         placeholder={placeholder}
-        value={inputValue}
-        validation={validation}
-        readOnly={readOnly || readOnlyInput}
-        disabled={disabled}
-        onChange={async (value) => {
-          await Actions.updateInputValue({ value, state })
-        }}
-        onBlur={async (e) => {
-          await Actions.onBlurInput({ value: e.target.value, state, selection })
-        }}
-        onFocus={async () => Actions.openDialog({ state })}
-        title={title}
+        ref={selectRef}
+        value={value}
```

Comment:
do we like to have the select as controlled or uncontrolled?

Reply:
controlled

Reply:
oki